### PR TITLE
Add FAQ entries for executor-per-GPU questions

### PR DESCRIPTION
**Pull request by jlowe**
Per the request at https://github.com/NVIDIA/spark-rapids/issues/1667#issuecomment-773378209, this adds FAQ entries discussing multiple GPUs per executor and multiple executors per GPU.

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -247,6 +247,25 @@ the I/O and starting the initial processing can suffer.  But if you have a lot o
...
+Many Spark setups avoid allocating too many concurrent tasks to the same executor, and often
+multiple executors are run per node on the cluster.  Therefore this feature has not been
+prioritized, as there has not been a compelling use-case that requires it.
+
+### Why are multiple executors per GPU not supported?
+
+There are multiple reasons why this a problematic configuration:
+- Apache Spark does not support scheduling a fractional number of GPUs to an executor
+- CUDA context switches between processes sharing a single GPU can be expensive
+- Each executor would have a fraction of the GPU memory available for processing
```
Comment (revans2) [inline docs/FAQ.md:267]: Do we want to add something about MIG?  It feels like it is related in some ways.

File: docs/FAQ.md [hunk 2]
```diff
@@ -247,6 +247,35 @@ the I/O and starting the initial processing can suffer.  But if you have a lot o
 cannot be done on the GPU, like complex UDFs, the more tasks you have the more CPU processing you
 can throw at it.
 
+### Why are multiple GPUs per executor not supported?
+
+The RAPIDS Accelerator only supports a single GPU per executor because that was a limitation of
+[RAPIDS cudf](https://github.com/rapidsai/cudf), the foundation of the Accelerator.  Basic support
```
Comment (gerashegalov) [inline docs/FAQ.md:253]: nit: check duplicate white space throughout

File: docs/FAQ.md [hunk 3]
```diff
@@ -247,6 +247,35 @@ the I/O and starting the initial processing can suffer.  But if you have a lot o
...
+for working with multiple GPUs has only recently been added to RAPIDS cudf, and there are no plans
```
Comment (gerashegalov) [inline docs/FAQ.md:254]: should we link to the cudf issue?

File: docs/FAQ.md (hunk 2 above)
Comment (jlowe) [reply docs/FAQ.md:253]: I don't think this is necessary.  Markdown reformats this anyway.

File: docs/FAQ.md (hunk 3 above)
Comment (jlowe) [reply docs/FAQ.md:254]: There are multiple issues, which one do I pick?

File: docs/FAQ.md [hunk 4]
```diff
@@ -247,6 +247,35 @@ the I/O and starting the initial processing can suffer.  But if you have a lot o
...
+- Apache Spark does not support scheduling a fractional number of GPUs to an executor
+- CUDA context switches between processes sharing a single GPU can be expensive
+- Each executor would have a fraction of the GPU memory available for processing
+
+### Is [Multi Instance GPU (MIG)](https://docs.nvidia.com/cuda/mig/index.html) supported?
+
+Yes, but it requires support from the underlying cluster manager to isolate the MIG GPU instance
+for each executor (e.g.: by setting `CUDA_VISIBLE_DEVICES` or other means).
+
+Note that MIG is not recommended for use with the RAPIDS Accelerator since it significantly
```
Comment (abellina) [inline docs/FAQ.md:274]: Might be nice to point this one out:

> No GPU to GPU P2P (either PCIe or NVLink) is supported

This is from: https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#app-considerations.

**Review by revans2 — APPROVED** [review]

File: docs/FAQ.md (hunk 2 above)
Comment (jlowe) [reply docs/FAQ.md]: I needed to update this for the MIG inter-GPU transfer, so I fixed the double-spaces in the parts that were added.

**Comment by jlowe** [comment]
build

**Review by abellina — APPROVED** [review]

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**